lib/cost_report.rb

```diff
@@ -24,23 +98,56 @@ def matching_clusters
     end
   end
 
-  def add_ht_item_to_freq_table(ht_item)
-    overlap = HtItemOverlap.new(ht_item)
-    overlap.matching_orgs.each do |organization|
-      @freq_table[organization.to_sym][overlap.matching_orgs.count] += 1
+  def total_hscore(member)
+    spm_total(member.to_sym) + mpm_total(member.to_sym) + ser_total(member.to_sym)
+  end
+
+  [:spm, :ser, :mpm].each do |format|
+    # HScore for a particular format
+    define_method "#{format}_total".to_sym do |member|
+      total = 0.0
+      freq_table[member.to_sym][format].each do |num_orgs, freq|
+        total += (1.0 / num_orgs * freq)
+      end
+      total
+    end
+
+    # Costs for a particular format
+    define_method "#{format}_costs".to_sym do |member|
+      public_send("#{format}_total", member) * cost_per_volume
     end
   end
 
-  def total_hscore
-    return @total_hscore unless @total_hscore.nil?
+  def total_ic_costs(member)
+    total_hscore(member) * cost_per_volume.to_f
+  end
 
-    @total_hscore = Hash.new {|hash, key| hash[key] = 0.0 }
-    @freq_table.each do |org, h_scores|
-      h_scores.each do |num_orgs, freq|
-        @total_hscore[org.to_sym] += (1.0 /num_orgs * freq)
-      end
+  def extra_per_member
+    total_ic_costs(:hathitrust).to_f / Services.ht_members.members.count
+  end
+
+  def total_cost_for_member(member)
+    total_ic_costs(member.to_s) + pd_cost_for_member(member.to_s) + extra_per_member
+  end
+
+  def to_tsv
+    report = []
+    report << ["member_id", "spm", "mpm", "ser", "pd", "weight", "extra", "total"].join("\t")
+    Services.ht_members.members.keys.sort.each do |member|
+      next unless organization.nil? || (member == organization.to_s)
+
+      report << [
+        member,
+        spm_costs(member),
```

Comment:
Maybe format all these as #.## on the way out? Is there a reason for the CSV to have arbitrary-precision floats in it?

Reply:
Only because arbitrary-precision is current behavior.